### ISSUE TITLE
Error using connectionstring

### ISSUE THREAD

**Issue by djmnz** [chucknorris/roundhouse#327]
Tryied to upgrade from 0.8.6 to 1.0.0-RC1

I can get consistently, the cause is just attempting to enter a connection string:

`.\rh.exe /cs="Data Source=.;Initial Catalog=<DATABASE>;Persist Security Info=Tru
e;User ID=<USERNAME>;Password=<PASSWORD>"`

Got the following error:

```

Error while trying to create the BuildPlan for Instance of roundhouse.databases.Database -- Lambda: build(IContext.GetInstance(), value(roundhouse.infrastructure.app.ApplicationConfiguraton+<>c__DisplayClass3_0).configuration_property_holder).
Please check the inner exception
1.) Container.GetInstance<roundhouse.databases.Database>()

StructureMap.StructureMapBuildPlanException: Error while trying to create the BuildPlan for Instance of roundhouse.databases.Database -- Lambda: build(IContext.GetInstance(), value(roundhouse.infrastructure.app.ApplicationConfiguraton+<>c__DisplayClass3_0).configuration_property_holder).
Please check the inner exception
1.) Container.GetInstance<roundhouse.databases.Database>()
 ---> System.TypeInitializationException: The type initializer for 'StructureMap.Building.TryCatchWrapper' threw an exception. ---> System.TypeLoadException: Could not load type '<Npgsql-IChunkingTypeHandler<System-Char\[\]>-Read>d__8' from assembly 'rh, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' because the parent does not exist.
   at System.Reflection.RuntimeAssembly.GetExportedTypes(RuntimeAssembly assembly, ObjectHandleOnStack retTypes)
   at System.Reflection.RuntimeAssembly.GetExportedTypes()
   at StructureMap.Building.TryCatchWrapper..cctor()
   --- End of inner exception stack trace ---
   at StructureMap.Building.TryCatchWrapper.WrapFunc[T](String message, Type returnType, Expression expression, IDescribed described)
   at StructureMap.Building.BuildPlan.ToDelegate()
   at StructureMap.Building.BuildPlan..ctor(Type pluginType, Instance instance, IDependencySource inner, Policies policies, IEnumerable`1 interceptors)
   at StructureMap.Pipeline.Instance.buildPlan(Type pluginType, Policies policies)
   --- End of inner exception stack trace ---
   at StructureMap.Pipeline.Instance.buildPlan(Type pluginType, Policies policies)
   at StructureMap.Pipeline.Instance.ResolveBuildPlan(Type pluginType, Policies policies)
   at StructureMap.BuildSession.BuildNewInSession(Type pluginType, Instance instance)
   at StructureMap.Pipeline.LazyLifecycleObject`1.CreateValue()
   at StructureMap.SessionCache.GetObject(Type pluginType, Instance instance, ILifecycle lifecycle)
   at StructureMap.SessionCache.GetDefault(Type pluginType, IPipelineGraph pipelineGraph)
   at StructureMap.Container.GetInstance[T]()
   at roundhouse.infrastructure.app.ApplicationConfiguraton.build_items_for_container(ConfigurationPropertyHolder configuration_property_holder)
   at roundhouse.infrastructure.app.ApplicationConfiguraton.build_the_container(ConfigurationPropertyHolder configuration_property_holder)
   at roundhouse.console.Program.set_up_configuration_and_build_the_container(String[] args, Mode mode)
   at roundhouse.console.Program.Main(String[] args)

```

If I just enter the rh.exe it shows the help message.

**Comment by BiggerNoise** [comment]
Thanks for reporting this. I have been very busy with family things and haven't been able to play with our 1.0 RC much, but I'll look into this as soon as I get a moment.

I'd highly suggest using the latest 0.9 if you're trying to get real work done; the RC in the 1.0.0 is, shall we say, highly optimistic.

**Comment by erikbra** [comment]
Looks like it's an ILMerge issue, the problem doesn't appear when running from non-ILMerged `rh.exe` (from the compile output folder). Now why does this happen?....

**Comment by BiggerNoise** [comment]
Thanks for looking into this. I assume that this is occurring in the full framework version (I don't think we're IL Mergine the core stuff yet)

**Comment by erikbra** [comment]
I had a look at this one... it's not too easy. The problem is that somewhere in StructureMap, it tries to enumerate all the exported type in the merged rh.exe. However, it seems ilmerge creates som strange types (named as in the error message), which it can't properly find. This is before starting resolving at all, in a static class. Unfortunately, StructureMap has been abandoned, and will not be updated. 

I tried using Lamar instead, but then I couldn't ilmerge it...

**Comment by BiggerNoise** [comment]
One thing we might want to consider is dropping structure map. It's a fine library, but a DI framework seems a bit overkill for our simple app.

I haven't looked into how bad this would be, but I doubt it would be too horrific.

**Comment by erikbra** [comment]
Fair point. It might be a bit drastic for solving this particular problem,
but, to think of it, a DI container doesn't serve a gigantic role in a
console app. The dependencies can easily be injected without using a DI
framework. We could look into it. The less dependencies, the better, imho.

2018-08-08 22:37 GMT+02:00 Andy Davis <notifications@github.com>:

**Comment by ferventcoder** [comment]
DI is a definite, whether or not you use a container is up to you. Testability at the unit level needs a good SOLID design - I would be a bit upset if the core design was lost considering how easily it was able to adapt back in the day. I'm sure that wasn't what you were mentioning here, but just wanted to throw that in.

**Comment by BiggerNoise** [comment]
Absolutely referring to the container, rather than DI itself. Going to take a look at the other P/Rs and see if we can temporize with Nject.

**Comment by erikbra** [comment]
This is solved in #335
